auth/auth.go

```diff
@@ -154,11 +148,14 @@ func (auth *Auth) Configure(config core.ServerConfig) error {
 			RootCAs:      trustStore.CertPool,
 			MinVersion:   core.MinTLSVersion,
 		}
-		if err = validator.SetValidatePeerCertificateFunc(tlsConfig); err != nil {
+		if err = auth.pkiValidator.AddTruststore(trustStore.Certificates()); err != nil {
```

Comment:
The truststore config is defined on root? Shouldn't this truststore initialization be performed by the PKI engine?

Reply:
We should probably also move `tls.go` and related code to this engine, but this and the config change are larger changes that need their own issue/PR